packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -38,11 +39,11 @@ abstract class Logger {
   ///
   /// [message] is the message to display to the user; [progressId] provides an ID which can be
   /// used to identify this type of progress (`hot.reload`, `hot.restart`, ...).
-  Status startProgress(String message, { String progressId });
+  Status startProgress(String message, { String progressId, bool expectedFastAction: false });
```

Comment:
I recommend flipping this around -- `expectSlowOperation`, default false. That would give us a better framing of the situation. Most operations should be fast. If they're not, it's a "slow operation", and it says so right there in the code, so we'll be more motivated to make it fast.